pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -643,6 +621,22 @@ func (s *DiscoveryServer) loadAssignmentsForClusterLegacy(push *model.PushContex
 	return l
 }
 
+// inSidecarScope checks if a given endpoint should be made available to a proxy
+func inSidecarScope(ep *model.IstioEndpoint, hostname model.Hostname, proxy *model.Proxy) bool {
```

Comment:
How often is this called ? Seems to have a big loop inside - any perf concerns ?

Reply:
Let me run some profiling on this and see if its showing up. It is called pretty often, I think you are right we should cache this

Reply:
Good catch, this was using up 10% of the CPU. caching this is pretty simple too, I will update the pr